Source/Core/Core/CoreTiming.h

```diff
@@ -58,6 +60,16 @@ struct Event
   u64 fifo_order;
   u64 userdata;
   EventType* type;
+
+  // Sort by time, unless the times are the same, in which case sort by the order added to the queue
+  constexpr auto operator<=>(const Event& other) const
+  {
+    return std::tie(time, fifo_order) <=> std::tie(other.time, other.fifo_order);
+  }
+  constexpr bool operator==(const Event& other) const
+  {
+    return std::tie(time, fifo_order) == std::tie(other.time, other.fifo_order);
+  }
```

Comment:
Do we need both `operator<=>` and `operator==`?

Reply:
User-defined `operator<=>` can only synthesize `<`, `<=`, `>`, and `>=`. Originally, it could generate all six comparison operators, but the EWG decided against it because some classes (e.g. `std::string`) could short-circuit evaluate their equality operators. Defaulted `operator<=>` still defines all six comparison operators, however.
We need all six comparison operator overloads to exist to use `std::ranges` comparison function objects, and I generally lean towards using those since they are newer and I assume offer improvements to justify the higher requirements.